parser/docker_cmd.go

```diff
@@ -114,7 +120,151 @@ func (b *BuildCmd) Run() (string, error) {
 			"error":    nil,
 		},
 	})
-	return image.ID, nil
+
+	var retID = image.ID
+
+	if opts.Squash {
+		libsquash.Verbose = true
+
+		var imageReader1, pipeWriter1 = io.Pipe()
+		var imageReader2, pipeWriter2 = io.Pipe()
+		var squashedImageReader, squashedImageWriter = io.Pipe()
+
+		defer func() {
+			for _, pipeReader := range []*io.PipeReader{imageReader1, imageReader2, squashedImageReader} {
+				pipeReader.Close()
+			}
+
+			for _, pipeWriter := range []*io.PipeWriter{pipeWriter1, pipeWriter2, squashedImageWriter} {
+				pipeWriter.Close()
+			}
+		}()
+
+		// exporting in a goroutine since the io.Pipe is synchronous
+		go func() {
```

Comment:
No, please! Comments welcome! Probably the most feedback is needed on the
PR currently out to libsquash. I having it using about 20% of memory it
originally used, but it still grows with the image size, which is not
ideal. There's also another issue where the layers don't seem to be
acknowledged by the daemon, which I'm guessing has to do with
the checksums.

On Sunday, January 4, 2015, Cole Gleason <notifications@github.com> wrote:

> In parser/docker_cmd.go
> <https://github.com/winchman/builder-core/pull/22#discussion-diff-22442379>
> :
>
> > +		var imageReader1, pipeWriter1 = io.Pipe()
> > +		var imageReader2, pipeWriter2 = io.Pipe()
> > +		var squashedImageReader, squashedImageWriter = io.Pipe()
> > +
> > +		defer func() {
> > +			for _, pipeReader := range []*io.PipeReader{imageReader1, imageReader2, squashedImageReader} {
> > +				pipeReader.Close()
> > +			}
> > +
> > +			for _, pipeWriter := range []*io.PipeWriter{pipeWriter1, pipeWriter2, squashedImageWriter} {
> > +				pipeWriter.Close()
> > +			}
> > +		}()
> > +
> > +		// exporting in a goroutine since the io.Pipe is synchronous
> > +		go func() {
>
> Ok cool, sorry! Just going through PRs and trying to review.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/winchman/builder-core/pull/22/files#r22442379>.
>